sabnzbd/decoder.py

```diff
@@ -239,7 +239,9 @@ def run(self):
                 # If the data needs to be written to disk due to full cache, this will be slow
                 # Causing the decoder-queue to fill up and delay the downloader
                 sabnzbd.ArticleCache.save_article(article, decoded_data)
-
+            elif not nzo.precheck:
+                # Nothing to save
+                article.on_disk = True
```

Comment:
@Safihre : Do you know if this can break anything? The intent is to let the assembler continue writing a file if an article is confirmed missing so that the cache doesn't fill up.